packages/website/private/contents/post-006.mdx

```diff
@@ -0,0 +1,60 @@
+---
+slug: type-safe-routing
+title: Type-safe routing
+description: Enjoy type safety for routing and navigation APIs in Waku projects.
+author: sophia
+date: 2024/12/10
+---
+
+Continuing our holiday retrospective of Waku’s recent features, today we’re excited to highlight type-safe routing. This addition brings compile-time safety to your routing and navigation logic while staying true to Waku’s minimalist principles.
+
+## Type-safe routing
+
+### Configuration
+
+While support for loose typing comes out of the box, you can opt into further type inference in TypeScript projects by adding `as const` assertions to the return of `getConfig` for each of your project’s routes.
+
+```tsx
+// ./src/pages/valid/[slug].tsx
+
+export default async function SomePage() { ... }
+
+export const getConfig = async () => {
+  const staticPaths = await getStaticPaths();
+
+  return {
+    render: 'static',
+    staticPaths,
+  } as const; // NEW!
+};
+
+const getStaticPaths = async () => { ... }
+```
+
+### Developer experience
+
+Afterwards enjoy type checks and IDE autocompletion for the `<Link>` component’s `to` prop as well as parameters of `useRouter()` methods like `router.push()`.
```

Comment:
We also export `PageProps` which can be used as `export const MyPage = (props: PageProps<'/link/[slug]'>) => {}` which would be nice to mention I think.